packages/blocks/src/api/validation.js

```diff
@@ -620,30 +620,30 @@ export function isEquivalentHTML( actual, expected ) {
  *
  * Logs to console in development environments when invalid.
  *
- * @param {string|Object} blockTypeOrName Block type.
- * @param {Object}        attributes      Parsed block attributes.
- * @param {string}        innerHTML       Original block content.
+ * @param {string|Object} blockTypeOrName      Block type.
+ * @param {Object}        attributes           Parsed block attributes.
+ * @param {string}        originalBlockContent Original block content.
  *
  * @return {boolean} Whether block is valid.
  */
-export function isValidBlockContent( blockTypeOrName, attributes, innerHTML ) {
+export function isValidBlockContent( blockTypeOrName, attributes, originalBlockContent ) {
 	const blockType = normalizeBlockType( blockTypeOrName );
-	let saveContent;
+	let generatedBlockContent;
 	try {
-		saveContent = getSaveContent( blockType, attributes );
+		generatedBlockContent = getSaveContent( blockType, attributes );
 	} catch ( error ) {
 		log.error( 'Block validation failed because an error occurred while generating block content:\n\n%s', error.toString() );
 		return false;
 	}
 
-	const isValid = isEquivalentHTML( innerHTML, saveContent );
+	const isValid = isEquivalentHTML( originalBlockContent, generatedBlockContent );
 	if ( ! isValid ) {
 		log.error(
-			'Block validation failed for `%s` (%o).\n\nExpected:\n\n%s\n\nActual:\n\n%s',
+			'Block validation failed for `%s` (%o).\n\nContent generated by `save` function:\n\n%s\n\nContent retrieved from post body:\n\n%s',
```

Comment:
There shouldn't be a concept of a "post" in this context. This logic applies for the post editor, but also any blocks parsing (e.g. upcoming widgets screen blocks editor). I realize the desire to improve messaging to developers. There may be alternatives which serve as an adequate compromise:

```
Markup produced by `save` function:

...

Markup parsed from content:

...
```